File: packages/models/lib/src/unit_ready_task.dart
import 'package:models/models.dart';

class UnitReadyTask implements Executable {
  final int unitId;
  final DateTime when;

  UnitReadyTask(this.unitId, this.when);

  @override
  int execute(Settlement settlement) {
    print('Inside UnitReadyTask task execute method.');
    final updatedList = List<int>.from(settlement.units);
    updatedList[unitId] += 1;
    settlement.units = updatedList;
    return 0;
  }

  @override
  DateTime get executionTime => when;
}

File: java_server/src/main/java/xyz/qruto/java_server/models/executables/UnitReadyTask.java
package xyz.qruto.java_server.models.executables;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import xyz.qruto.java_server.entities.SettlementEntity;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UnitReadyTask implements Executable {
    private int unitId;
    private LocalDateTime when;

    @Override
    public int execute(SettlementEntity settlement) {
        System.out.println("Inside UnitReadyTask task execute method.");
        var oldAmount = settlement.getUnits().get(unitId);
        settlement.getUnits().set(unitId, oldAmount + 1);
        return 0;
    }

    @Override
    public LocalDateTime getExecutionTime() {
        return when;
    }


}

